Integrated-Windows-Auth-Flow/src/main/java/TokenCacheAspect.java

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import com.microsoft.aad.msal4j.ITokenCacheAccessAspect;
+import com.microsoft.aad.msal4j.ITokenCacheAccessContext;
+
+import java.net.URI;
+import java.net.URL;
+import java.nio.file.*;
+import java.util.HashMap;
+import java.util.Map;
+
+public class TokenCacheAspect implements ITokenCacheAccessAspect {
```

Comment:
this implementation on ITokenCacheAccessAspect does not make much sense.

1. Token cache is initialized from file with "dummy data", after it, in-memory "data" defined in TokenCacheAspect 
is used. So file is used to read "dummy data" and it is never updated. That does not make sense.

2. Using in-memory cache defined in TokenCacheAspect is pointless, because MSAL application has its own in-memory token cache.


The whole point

Reply:
When I wrote this originally I as hesitant to write customer's credentials to a file - I imagine that a significant percentage of people who are just starting to use MSAL won't realize it that tokens are being written to a file and might then just casually run the sample and then just leave that Json file with their credentials sitting there. To avoid this, I Instead I left a comment explaining where you could write code to update the file, if so they so chose to do it. 

If we decide to make this change (writing updates to the file by default), we we should clearly document that the sample will writes tokens to the JSON file, and explain the consequences of having those tokens leaked.

Reply:
I do not think we should show how to implement persistent cache in scope of this sample, it is quite complicated and should :
- demonstrate real use case ( as I explained current implementation does not make sense )
- cover security
- concurrency